addons/medical_engine/functions/fnc_getHitpointArmor.sqf

```diff
@@ -0,0 +1,85 @@
+#include "script_component.hpp"
+/*
+ * Author: Pterolatypus
+ * Checks a unit's equipment to calculate the total armor on a hitpoint
+ *
+ * Arguments:
+ * 0: Unit <OBJECT>
+ * 1: Hitpoint <STRING>
+ *
+ * Return Value:
+ * Total armor for the given hitpoint <ARRAY>
+ * 0: Armor <NUMBER>
+ * 1: ExplosionShielding <NUMBER>
+ *
+ * Example:
+ * [player, "HitChest"] call ace_medical_engine_fnc_getHitpointArmor
+ *
+ * Public: No
+ */
+params ["_unit", "_hitpoint"];
+
+// Helper function to check cache and/or read config
+private _fnc_getItemArmor = {
+    params ["_item", "_hitpoint"];
+
+    private _key = format ["%1$%2", _item, _hitpoint];
+    private _armorValues = GVAR(armorCache) getVariable _key;
+
+    if (isNil "_armorValues") then {
+
+        TRACE_2("Cache miss", _item, _hitpoint);
+        if (_item isEqualTo "") exitWith {
+            _armorValues = [0,0];
+            GVAR(armorCache) setVariable [_key, _armorValues];
+        };
+
+        private _itemInfo = configFile >> "CfgWeapons" >> _item >> "ItemInfo";
+
+        if (getNumber (_itemInfo >> "type") == TYPE_UNIFORM) then {
+            private _unitCfg = configFile >> "CfgVehicles" >> getText (_itemInfo >> "uniformClass");
+            private _baseValues = GVAR(armorCache) getVariable (_item + "$$base");
+
+            if (isNil "_baseValues") then {
+                _baseValues = [getNumber (_unitCfg >> "armor"), getNumber (_unitCfg >> "explosionShielding")];
+                GVAR(armorCache) setVariable [(_item + "$$base"), _baseValues];
+            };
+
+            private _entry = _unitCfg >> "HitPoints" >> _hitpoint;
+            _armorValues = [
+                (_baseValues select 0) * getNumber (_entry >> "armor"),
+                (_baseValues select 1) * getNumber (_entry >> "explosionShielding")
+            ];
```

Comment:
I have entangled this part a bit:
```sqf
         if (getNumber (_itemInfo >> "type") == TYPE_UNIFORM) then {
            private _config = configFile >> "CfgVehicles" >> getText (_itemInfo >> "uniformClass");
            private _entry = _config >> "HitPoints" >> _hitpoint;

            private _key = format ["%1$$base", _item];
            GVAR(armorCache) getVariable _key params ["_armor", "_explosion"];

            if (isNil "_armor") then {
                _armor = getNumber (_config >> "armor");
                _explosion = getNumber (_config >> "explosionShielding");
                GVAR(armorCache) setVariable [_key, [_armor, _explosion]];
            };

            _armorValues = [
                _armor * getNumber (_entry >> "armor"),
                _explosion * getNumber (_entry >> "explosionShielding")
            ];
        };
```

But then it occured to me that this is all inside a cache anyway. The lookups are minimal.
Really this all should just be:
```sqf
         if (getNumber (_itemInfo >> "type") == TYPE_UNIFORM) then {
            private _uniformClass = configFile >> "CfgVehicles" >> getText (_itemInfo >> "uniformClass");
            private _hitpointEntry = _uniformClass >> "HitPoints" >> _hitpoint;

            _armorValues = [
                getNumber (_uniformClass >> "armor") * getNumber (_hitpointEntry >> "armor"),
                getNumber (_uniformClass >> "explosionShielding") * getNumber (_hitpointEntry >> "explosionShielding")
            ];
        };
```